tests/mock_crust/client_restrictions.rs

```diff
@@ -151,17 +152,26 @@ fn resend_parts_on_exceeding_limit() {
     }
     poll_and_resend(&mut nodes, &mut clients);
 
-    // `MAX_PARTS / 2` parts will be resent from client.
-    let expect_sent_parts = (2 * MAX_PARTS) as u64;
-    assert_eq!(
-        clients[0].inner.get_user_msg_parts_count(),
-        expect_sent_parts
+    // NOTE: these calculations are somewhat approximate – in reality, the user message
+    // parts seem to be around 20,363 bytes each, not 20,480, so an extra one can
+    // sometimes be crammed into `SOFT_CAPACITY`.
+    let total_data_parts = num_immutable_data * MAX_PARTS as u64;
+    let parts_allowed_first_time = SOFT_CAPACITY / MAX_PART_LEN as u64;
+    let parts_retried = total_data_parts - parts_allowed_first_time;
+    assert!(parts_retried > 0);
```

Comment:
I removed this assertion entirely as it was somewhat redundant. I wrote it initially just to check that I was calculating a meaningful non-zero value for `parts_retried`.